### DISCUSSION TITLE
Welcome to Ducobox-ESPEasy-Plugin Discussions!

### DISCUSSION THREAD

**Discussion by arnemauer** [arnemauer/Ducobox-ESPEasy-Plugin#80] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Theoreet** [comment]
Is there some progress with a ESPHome plugin, or a news bin file together with the newest ESPEasy version ?

**Comment by arnemauer** [comment]
Ik ben de afgelopen weken erg druk bezig geweest om de code volledig om te bouwen. Ik ben nu de laatste zaken aan het testen en hoop met 1-2 weken een nieuwe firmware online te kunnen zetten.

Ik zal dan ook een nieuwe esp versie gebruiken waarin een aantal grote wifi problemen zijn opgelost.

**Reply by rhvs** [reply]
Al een idee wanneer er wat getest kan worden? Laatste weken krijgt de ducobox om een of andere reden random een manual signaal voor "1" danwel "low", wat best gek is. Ik wilde de gateway al een keer opnieuw flashen, maar als je dan toch bijna een nieuwe versie hebt wacht ik wel even :)

**Reply by arnemauer** [reply]
Ik zal vanavond een .bin posten. Dit is echter wel een niet geteste beta versie! :)

**Reply by arnemauer** [reply]
Hier een beta versie met veel aanpassingen in de RF plugin.
Ik zal kijken of het lukt om vanavond nog een versie te compilen op een nieuwe espeasy versie.

[ESP_Easy_mega_20211122_custom_ESP8266_4M1M.zip](https://github.com/arnemauer/Ducobox-ESPEasy-Plugin/files/7584048/ESP_Easy_mega_20211122_custom_ESP8266_4M1M.zip)

**Reply by arnemauer** [reply]
En een beta versie gebaseerd op ESP_Easy_mega_20211122:
[ESP_Easy_mega_20211122_custom_ESP8266_4M1M.zip](https://github.com/arnemauer/Ducobox-ESPEasy-Plugin/files/7584191/ESP_Easy_mega_20211122_custom_ESP8266_4M1M.zip)

**Reply by rhvs** [reply]
CPlugin 5 not included in build - dan heb ik zeker de verkeerde zip gebruikt?

**Reply by arnemauer** [reply]
Welke controllers gebruik je? Het zou kunnen dat hij niet in deze build zit.

**Reply by rhvs** [reply]
alleen Home Assistant via OpenHAB MQTT

**Reply by rhvs** [reply]
verdere feedback - er zit een standaard Easy ESP ww op het wifi - waar het hiervoor nog open stond. Kleinigheidje, maar ik moest even zoeken wat het std ww was.

**Reply by arnemauer** [reply]
Was een build_flag vergeten toe te voegen waardoor niet alle controllers en settings goed stonden. Hier een nieuwe volledige build :
[ESP_Easy_mega_20211123_normal_ESP8266_4M1M_VCC.zip](https://github.com/arnemauer/Ducobox-ESPEasy-Plugin/files/7590618/ESP_Easy_mega_20211123_normal_ESP8266_4M1M_VCC.zip)
)

**Reply by rhvs** [reply]
Werkt wel, maar de bug die ik had (bij reboot of wifi issues) die is er nog steeds. Als ik reboot, dan gaat er een manual 1 (vorige FW) of 3 (huidige FW) naar de ducobox. Uitschakelen van controller zorgt ervoor dat dit niet gebeurt bij reboot van de GW, maar verder zou ik niet weten waar ik het moet zoeken..

**Reply by arnemauer** [reply]
Wat heb je aan die controller gekoppeld? HA of domoticz?
Is het de gateway zelf die dat commando verstuurd of zou het via een domotica systeem kunnen komen? 
Zou je een log kunnen posten vanaf een reboot?

**Reply by rhvs** [reply]
Zeker, niet vanaf reboot maar wel net zo lang gewacht tot hij spontaan weer op high ging staan... hieronder een stuk log daaromheen. Hopelijk kan je er wat mee :)

23880483: [P150] DUCO RF GW: UPDATE by PLUGIN_READ
23880483: [P150] DUCO RF GW: State: 0.00
23905153: [P150] DUCO RF GW: 23905152 - RECEIVED message: SRC:2; DEST:1; ORG.SRC:2; ORG.DEST:1; Ntwrk:000163c4;Type:7; Bytes:16; Counter:
23905153: [P150] DUCO RF GW: 23905152 - 50,00,22,3D,00,59,00,
23905154: [P150] DUCO RF GW: 23905152 - RECEIVED message: SRC:1; DEST:2; ORG.SRC:1; ORG.DEST:2; Ntwrk:000163c4;Type:6; Bytes:9; Counter:1
23905154: [P150] DUCO RF GW: 23905153 -
23925234: MQTT : Connection lost, state: Disconnected
23925317: MQTT : Connected to broker with client ID: VENTILATION_GATEWAY
23925320: Subscribed to: VENTILATION_GATEWAY/#
23940394: WD : Uptime 390 ConnectFailures 0 FreeMem 8632 WiFiStatus 3 ESPeasy internal wifi status: Conn. IP Init
24020607: [P150] DUCO RF GW: 23992510 - No free space in InboxQ, dropping message;
24020607: [P150] DUCO RF GW: 23992516 - No free space in InboxQ, dropping message;
24020607: [P150] DUCO RF GW: 24017535 - No free space in InboxQ, dropping message;
24020608: [P150] DUCO RF GW: 24017540 - No free space in InboxQ, dropping message;
24020608: [P150] DUCO RF GW: 24020604 - RECEIVED message: SRC:2; DEST:1; ORG.SRC:2; ORG.DEST:1; Ntwrk:000163c4;Type:7; Bytes:16; Counter:
24020608: [P150] DUCO RF GW: 24020605 - 50,00,22,3C,00,59,00,
24020609: [P150] DUCO RF GW: 24020605 - RECEIVED message: SRC:1; DEST:2; ORG.SRC:1; ORG.DEST:2; Ntwrk:000163c4;Type:6; Bytes:9; Counter:1
24020609: [P150] DUCO RF GW: 24020605 -
24020609: [P150] DUCO RF GW: 24020606 - RECEIVED message: SRC:2; DEST:1; ORG.SRC:2; ORG.DEST:1; Ntwrk:000163c4;Type:7; Bytes:16; Counter:
24020610: [P150] DUCO RF GW: 24020606 - 50,00,22,3D,00,59,00,
24020610: [P150] DUCO RF GW: 24020606 - RECEIVED message: SRC:1; DEST:2; ORG.SRC:1; ORG.DEST:2; Ntwrk:000163c4;Type:6; Bytes:9; Counter:1
24020611: [P150] DUCO RF GW: 24020606 -
24080778: [P150] DUCO RF GW: 24080776 - RECEIVED message: SRC:2; DEST:1; ORG.SRC:2; ORG.DEST:1; Ntwrk:000163c4;Type:7; Bytes:16; Counter:
24080778: [P150] DUCO RF GW: 24080776 - 50,00,22,3D,00,59,00,
24080779: [P150] DUCO RF GW: 24080776 - RECEIVED message: SRC:1; DEST:2; ORG.SRC:1; ORG.DEST:2; Ntwrk:000163c4;Type:6; Bytes:9; Counter:1
24080779: [P150] DUCO RF GW: 24080776 -
24080781: [P150] DUCO RF GW: UPDATE by PLUGIN_READ
24080782: [P150] DUCO RF GW: State: 0.00
24080906: Command: VENTMODE
24080907: [P150] DUCO RF GW: Sent command for 'VENTMODE' & 'high' to DUCO unit
24080915: [P150] DUCO RF GW: 24080907 - requestVentilationMode();
24080916: [P150] DUCO RF GW: 24080907 - sendVentilationModeMessage();
24080916: [P150] DUCO RF GW: 24080908 - SENT message: SRC:4; DEST:1; ORG.SRC:4; ORG.DEST:1; Ntwrk:000163c4;Type:7; Bytes:18; Counter:5;
24080916: [P150] DUCO RF GW: 24080908 - 00,07,00,01,63,C4,20,48,15,00,60,00,22,00,0E,6E,00,0B,
24080917: [P150] DUCO RF GW: 24080915 - Start waiting for ack...
24080962: [P150] DUCO RF GW: 24080962 - Q0 hassent:1; waitforack:1;ackreceived:0; timer: 24080915; sendTries: 0
24081271: [P150] DUCO RF GW: 24081262 - CheckforAck: still waiting for ACK. Sending message again...
24081272: [P150] DUCO RF GW: 24081263 - SENT message: SRC:4; DEST:1; ORG.SRC:4; ORG.DEST:1; Ntwrk:000163c4;Type:7; Bytes:18; Counter:5;
24081272: [P150] DUCO RF GW: 24081263 - 00,07,00,01,63,C4,20,48,15,00,60,00,22,00,0E,6E,00,0B,
24081363: [P150] DUCO RF GW: 24081362 - Q0 hassent:1; waitforack:1;ackreceived:0; timer: 24081270; sendTries: 1
24081462: [P150] DUCO RF GW: 24081462 - Q0 hassent:1; waitforack:1;ackreceived:0; timer: 24081270; sendTries: 1
24081562: [P150] DUCO RF GW: 24081562 - Q0 hassent:1; waitforack:1;ackreceived:0; timer: 24081270; sendTries: 1
24081670: [P150] DUCO RF GW: 24081662 - Q0 hassent:1; waitforack:1;ackreceived:0; timer: 24081270; sendTries: 1
24081670: [P150] DUCO RF GW: 24081662 - CheckforAck: check if 300ms is verstreken
24081670: [P150] DUCO RF GW: 24081662 - CheckforAck: still waiting for ACK. Sending message again...
24081671: [P150] DUCO RF GW: 24081662 - SENT message: SRC:4; DEST:1; ORG.SRC:4; ORG.DEST:1; Ntwrk:000163c4;Type:7; Bytes:18; Counter:5;
24081671: [P150] DUCO RF GW: 24081663 - 00,07,00,01,63,C4,20,48,15,00,60,00,22,00,0E,6E,00,0B,
24081762: [P150] DUCO RF GW: 24081762 - Q0 hassent:1; waitforack:1;ackreceived:0; timer: 24081669; sendTries: 2
24081862: [P150] DUCO RF GW: 24081862 - Q0 hassent:1; waitforack:1;ackreceived:0; timer: 24081669; sendTries: 2
24081962: [P150] DUCO RF GW: 24081962 - Q0 hassent:1; waitforack:1;ackreceived:0; timer: 24081669; sendTries: 2
24082103: [P150] DUCO RF GW: 24082095 - Q0 hassent:1; waitforack:1;ackreceived:0; timer: 24081669; sendTries: 2
24082104: [P150] DUCO RF GW: 24082096 - CheckforAck: check if 300ms is verstreken
24082104: [P150] DUCO RF GW: 24082096 - CheckforAck: still waiting for ACK. Sending message again...
24082105: [P150] DUCO RF GW: 24082096 - SENT message: SRC:4; DEST:1; ORG.SRC:4; ORG.DEST:1; Ntwrk:000163c4;Type:7; Bytes:18; Counter:5;
24082105: [P150] DUCO RF GW: 24082096 - 00,07,00,01,63,C4,20,48,15,00,60,00,22,00,0E,6E,00,0B,
24082115: [P150] DUCO RF GW: 24082114 - RECEIVED message: SRC:1; DEST:4; ORG.SRC:1; ORG.DEST:4; Ntwrk:000163c4;Type:6; Bytes:9; Counter:5
24082116: [P150] DUCO RF GW: 24082114 -
24082116: [P150] DUCO RF GW: 24082114 - Received messagetype: ACK
24082116: [P150] DUCO RF GW: 24082114 - hassent:1; waitforack:1;ackreceived:0; counter: 5
24082116: [P150] DUCO RF GW: 24082114 - HAS SENT TRUE
24082117: [P150] DUCO RF GW: 24082115 - waitfor ACK = TRUE en ackreceived= FALSE
24082117: [P150] DUCO RF GW: 24082115 - Received ACK for a message we've sent.
24084816: [P150] DUCO RF GW: 24084805 - Send sendConfirmationNewVentilationMode
24084816: [P150] DUCO RF GW: 24084806 - SENT message: SRC:4; DEST:1; ORG.SRC:4; ORG.DEST:1; Ntwrk:000163c4;Type:7; Bytes:18; Counter:6;
24084816: [P150] DUCO RF GW: 24084806 - 00,07,00,01,63,C4,20,48,16,00,60,00,22,00,06,6E,00,00,
24084817: [P150] DUCO RF GW: 24084813 - send response
24084817: [P150] DUCO RF GW: 24084813 - Start waiting for ack...
24084821: [P150] DUCO RF GW: 24084820 - RECEIVED message: SRC:1; DEST:4; ORG.SRC:1; ORG.DEST:4; Ntwrk:000163c4;Type:6; Bytes:9; Counter:6
24084822: [P150] DUCO RF GW: 24084820 -
24084822: [P150] DUCO RF GW: 24084820 - Received messagetype: ACK
24084822: [P150] DUCO RF GW: 24084820 - hassent:1; waitforack:1;ackreceived:0; counter: 6
24084822: [P150] DUCO RF GW: 24084820 - HAS SENT TRUE
24084822: [P150] DUCO RF GW: 24084821 - waitfor ACK = TRUE en ackreceived= FALSE
24084823: [P150] DUCO RF GW: 24084821 - Received ACK for a message we've sent.
24085744: [P150] DUCO RF GW: ventilation mode changed -> UPDATE by PLUGIN_ONCE_A_SECOND
24085745: [P150] DUCO RF GW: State: 3.00

**Reply by rhvs** [reply]
Net weer een reboot van de GW - kan je iets met onderstaande output?

Uptime: | 0 days 0 hours 0 minutes
Load: | 100.00% (LC=3)
CPU Eco Mode: | false
Boot: | Exception (30)
Reset Reason: | Exception
Last Action before Reboot: | SystemEventQueue: timer, id: 66603

**Reply by rhvs** [reply]
OK - update for the archive... it's fixed! The issue was some retained messages which I deleted with MQTT Explorer. So now nothing has the retain flag and I'm not sure if that's ok, but at least the machine isn't going to high ventilation at reboot!

**Reply by Theoreet** [reply]
komt deze nog uit beta ?

**Reply by arnemauer** [reply]
Morgen ;)

**Comment by Theoreet** [comment]
Goedenavond Arne

Ben erg benieuwd naar de stand van je nieuwe code

Heb je iets wat ik kan testen bijvoorbeeld ?

Gr


Op ma 1 nov. 2021 22:28 schreef Arne Mauer ***@***.***>:

**Comment by mkampstra** [comment]
Ah, ik was al op zoek naar nieuwe update, maar die is onderweg... 
Wacht ik nog even geduldig...

**Comment by NielsDijkstra** [comment]
Ik heb net mijn Ducobox werkend gekregen in combinatie met Domoticz met de stappen van deze Wiki, erg handig, thx!
kleine opmerking t.o.v. de manual:
Bij mijn gateway heb ik alleen niet (meer) de optie om "send to controller" uit te zetten, dus Domoticz krijgt elke 60 seconden een update.
Ik heb daarom de url in de Rules van een extra optie voorzien om de loop te voorkomen. "&parsetrigger=false" Dit zorgt ervoor dat de sensor wel geupdate wordt naar de laatste status, maar dat er geen trigger wordt uitgevoerd.
Bij mij werkt dit goed.
Voordeel hiervan is ook dat Domoticz dus wel altijd de actuele status heeft, ook als de instellingen vanaf een andere remote worden aangepast.  
voorbeeld:
`SendToHTTP xx.xx.xx.xx,80,/json.htm?type=command&param=udevice&idx=YYY&nvalue=0&svalue=50&parsetrigger=false `

**Comment by funnyfact** [comment]
Ik heb een Duco rf 325, ik heb een aantal pogingen gedaan tot koppelen, maar krijg geen enkele reactie van de ventilatie box. 
Volgens de log file wordt er een join verstuurd. Ik zie ook een uitslag als ik het radio verkeer bekijk via een RTL Dongel. 

Ik vraag me af zou het moeten werken met een Duco rf 325? 

984345: FILE : Saved security.dat
984345: SaveToFile: free stack after: 2992
984357: [P150] DUCO RF GW: Extra Settings PLUGIN_150 loaded
984369: [P150] DUCO RF GW: Values set from config. DeviceID: 0, networkId: 0000, radio power: Radio power 10,3 dBm (0xC1)
984372: [P150] DUCO RF GW: CC1101 868Mhz transmitter initialized
988261: Webserver args: 0: 'cmd' length: 4
988271: Command: JOIN
988272: [P150] DUCO RF GW: Sent command for 'join' to DUCO unit
988278: [P150] DUCO RF GW: 988272 - SendJoinPacket()
988279: [P150] DUCO RF GW: 988272 - 00,01,00,00,7C,3E,08,02,01,FF,0C,
988279: [P150] DUCO RF GW: 988278 - Joinpacket sent. DucoDeviceState = ducoDeviceState_join1
990281: Webserver args: 0: 'cmd' length: 4
990287: Command: JOIN
990288: [P150] DUCO RF GW: Sent command for 'join' to DUCO unit
990294: [P150] DUCO RF GW: 990288 - SendJoinPacket()
990295: [P150] DUCO RF GW: 990289 - 00,01,00,00,7C,3E,08,02,01,FF,0C,
990295: [P150] DUCO RF GW: 990294 - Joinpacket sent. DucoDeviceState = ducoDeviceState_join1
991546: Webserver args: 0: 'cmd' length: 4

**Comment by bartlauwers53** [comment]
Ik heb een apart verschijnsel: als ik mijn box op één van de drie tijdelijke standen zet, geeft de countdown plugin in de webconsole van de ducobox soms 295 weer, en dan 0, en vaak ook waardes als 4720. Deze waardes dus, in plaats van wat ik zou vewachten dat ie ergens rond de 900 seconden (is 15 minuten) gaat aftellen(updaten per minuut, of wat je instelt, ik weer dat dat niet realtime te zien is, wat prima is). Dit zag ik doordat in HA deze waardes ook keurig doorgegeven worden. Overigens komen die waardes ook regelmatig voor als ik de box gewoon op auto heb staan. 
Heeft iemand hier een verklaring voor? Issue van aanmaken?

**Comment by Theoreet** [comment]
Toevallig 😁😉

Op di 29 mrt. 2022 23:34 schreef Arne Mauer ***@***.***>:

**Comment by ericsteffann-fontys** [comment]
@arnemauer Heb je nog een gateway te koop? Ik heb het Google formulier ingevuld, maar niets meer vernomen.

**Reply by arnemauer** [reply]
Ik wacht nog op levering van onderdelen. Ik hoop dat die binnen een week binnen zijn. Daarna duurt het nog ongeveer 3 weken voordat ik ze pcb's in huis heb.

**Comment by psuurbach** [comment]
Arne, 

Vandaag de gateway ontvangen, geïnstalleerd met door jou opgenomen stappen en alles werkte direct! ook Homeasisstant ontvangt de MQTT berichten. klasse!!! een zeer nette oplossing. Ik ben echter benieuwd of iemand al zijn configuratie uit Home assistant kan delen. 

Configureer de sensor data in configurations.yaml:

Configureer de switches in configurations.yaml:

Configureer de cards in Lovelace.

Of kan je me een eind op weg helpen hiermee?

Met dank, Met groet,

Paul

**Comment by Yihaaa73** [comment]
> ## 👋 Welcome!
> We’re using Discussions as a place to connect with other members of our community. We hope that you:
> 
> * Ask questions you’re wondering about.
> * Share ideas.
> * Engage with other community members.
> * Welcome others and are open-minded. Remember that this is a community we
>   build together 💪.
> 
> To get started, comment below with an introduction of yourself and tell us about what you do with this community.

Komt er nog een Gateway binnenkort op voorraad? Ik hoop het wel. Schitterend project!

**Reply by Yihaaa73** [reply]
Komt de Gateway nog op voorraad? Ik wil er graag 1 bestellen.

**Reply by Maarten71** [reply]
En ik ook graag.

**Reply by bartlauwers53** [reply]
Misschien helpt het als we @arnemauer even taggen maar de plugin lijkt inmiddels niet heel actief beheerd te worden.

**Reply by arnemauer** [reply]
Ik heb een aantal testpcb's gemaakt met een nieuwe antenne. Deze antenne moet nog getuned worden. De nanoVNA en de onderdelen voor het tunen van de antenne zijn deze week binnengekomen. Ik hoop ergens in de komende 2 weken tijd te hebben om de antenne te gaan tunen. Als dat lukt, ik heb er geen ervaring in :), dan heb ik een paar testexemplaren voor jullie.

**Reply by Theoreet** [reply]
Ik test eventueel graag mee als het kan helpen/nog iets over is

**Reply by Maarten71** [reply]
Prima om mee te testen. Wil evt ook helpen om doc aan te passen.

**Comment by Maarten71** [comment]
@arnemauer , heb je nog een update over de nieuwe test pcb's?

**Comment by leroyraw** [comment]
Hi @arnemauer , komen er nog usb stekkers of gateways op voorraad? Zag even geen andere plek om deze vraag te stellen. Fantastisch project, zou het graag proberen!
